node/src/bootstrap/bootstrap_attempt.rs

```diff
@@ -254,3 +256,38 @@ pub struct BootstrapExited {
     pub total_blocks: String,
     pub duration: String,
 }
+
+#[derive(Clone)]
+pub struct BootstrapCallbacks {
+    pub(crate) bootstrap_started_observer:
+        Arc<Mutex<Vec<Arc<dyn Fn(BootstrapCallbackData) + Send + Sync>>>>,
+    pub(crate) bootstrap_ended_observer:
+        Arc<Mutex<Vec<Arc<dyn Fn(BootstrapCallbackData) + Send + Sync>>>>,
+}
+
+impl BootstrapCallbacks {
+    pub(crate) fn new() -> Self {
+        Self {
+            bootstrap_started_observer: Arc::new(Mutex::new(Vec::new())),
+            bootstrap_ended_observer: Arc::new(Mutex::new(Vec::new())),
+        }
+    }
+}
+
+pub struct BootstrapCallbackData {
+    pub(crate) id: String,
+    pub(crate) mode: BootstrapMode,
+    pub(crate) total_blocks: u64,
+    pub(crate) duration: Duration,
+}
+
+impl BootstrapCallbackData {
+    fn new(id: String, mode: BootstrapMode, total_blocks: u64, duration: Duration) -> Self {
+        Self {
+            id,
+            mode,
+            total_blocks,
+            duration,
+        }
+    }
```

Comment:
I would prefer to not have a new function here because we have many params (4 fields) and readability is better if we assign the fields by name then:
`
BootstrapCallbackData {
  id: ...,
  mode: ...
}`